bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/validation/SyncTableValidationOnlyJob.java

```diff
@@ -0,0 +1,1102 @@
+          job.getCredentials(), new Path[] {sourceHashDir}, getConf());
+    }
+
+    HashTable.TableHash tableHash = HashTable.TableHash.read(getConf(), sourceHashDir);
+    String tableName = BigtableTableHashAccessor.getTableName(tableHash);
+    int numHashFiles = BigtableTableHashAccessor.getNumHashFiles(tableHash);
+    String hashDataDir = BigtableTableHashAccessor.getHashDataDir();
+    String hashOutputDataFilePrefix = BigtableTableHashAccessor.getHashOutputDataFilePrefix();
+    List<ImmutableBytesWritable> partitions = BigtableTableHashAccessor.getPartitions(tableHash);
+
+    LOG.info("Read source hash manifest: " + tableHash);
+    LOG.info("Read " + partitions.size() + " partition keys");
+    if (!tableName.equals(sourceTableName)) {
+      LOG.warn(
+          "Table name mismatch - manifest indicates hash was taken from: "
+              + tableName
+              + " but job is reading from: "
+              + sourceTableName);
+    }
+    if (numHashFiles != partitions.size() + 1) {
+      throw new RuntimeException(
+          "Hash data appears corrupt. The number of of hash files created"
+              + " should be 1 more than the number of partition keys.  However, the manifest file "
+              + " says numHashFiles="
+              + numHashFiles
+              + " but the number of partition keys"
+              + " found in the partitions file is "
+              + partitions.size());
+    }
+
+    Path dataDir = new Path(sourceHashDir, hashDataDir);
+    int dataSubdirCount = 0;
+    for (FileStatus file : fs.listStatus(dataDir)) {
+      if (file.getPath().getName().startsWith(hashOutputDataFilePrefix)) {
+        dataSubdirCount++;
+      }
+    }
+
+    if (dataSubdirCount != numHashFiles) {
+      throw new RuntimeException(
+          "Hash data appears corrupt. The number of of hash files created"
+              + " should be 1 more than the number of partition keys.  However, the number of data dirs"
+              + " found is "
+              + dataSubdirCount
+              + " but the number of partition keys"
+              + " found in the partitions file is "
+              + partitions.size());
+    }
+
+    job.setJarByClass(HashTable.class);
+    jobConf.set(SOURCE_HASH_DIR_CONF_KEY, sourceHashDir.toString());
+    jobConf.set(SOURCE_TABLE_CONF_KEY, sourceTableName);
+    jobConf.set(TARGET_TABLE_CONF_KEY, targetTableName);
+    if (sourceZkCluster != null) {
+      jobConf.set(SOURCE_ZK_CLUSTER_CONF_KEY, sourceZkCluster);
+      initCredentialsForHBase(sourceZkCluster, job);
+    }
+    if (sourceBigtableProjectId != null && sourceBigtableInstance != null) {
+      if (sourceZkCluster != null)
+        LOG.warn(
+            "sourceZkCluster config("
+                + sourceZkCluster
+                + ") overriden with sourceBigtableProjectId("
+                + sourceBigtableProjectId
+                + "), sourceBigtableInstance("
+                + sourceBigtableInstance
```

Comment:
I think we should fail if they are both set. This seems like an invalid input and rather than giving users a false result (they wanted to validate against hbase, but instead verified bigtable with bigtable). Rather than interpreting what user wants, we should fail fast and let user tell us explicitly what they want.